### PR TITLE
re-adds slapping action(this one is different from the boring emote)

### PR DESCRIPTION
**Pull request by Couls**
## About The Pull Request
For when someone pats you on the head

## Why It's Good For The Game
Hey man I don't wanna do this.. I said don't come at me, Don't come at me!
ezgif com-video-to-gif (1)

## Changelog
:cl:
add: You can slap people by aiming at the mouth with disarm intent
/:cl:

### CODE REVIEW

**Comment by Borisvanmemes** [comment]
We still have slapping though via *slap though, it's jank and bad but we have it.

**Comment by Couls** [comment]
You could've emoted headpats too it's only fair slapping gets an action as well

**Comment by Tharcoonvagh** [comment]
_what are you talking about_

**Comment by Tharcoonvagh** [comment]
Shit they merged it

**Comment by Angustmeta** [comment]
Shouldn't add emotes to combat combinations (intent + aim at bodypart). Adds risk of accidental use in actual combat. Having the slap emote is better imo.

**Comment by ATH1909** [comment]
> Shouldn't add emotes to combat combinations (intent + aim at bodypart). Adds risk of accidental use in actual combat. Having the slap emote is better imo.

Plus, the slap emote lets xenos and monkeys and such slap too.

You have not known true fear until you've seen a xenomorph queen neurospit you, then stride towards you menacingly while slapping her own ass.

**Comment by Misdoubtful** [comment]
More escalation content in-between the options of talking and brawling GOOD. More true to scale escalation of violence GOOD.

**Comment by anconfuzedrock** [comment]
for people who are confused this is basically a meme pr
I originally coded slapping when headpatting was originally added and then both were removed
slapping as it is now is functionally identical except far less suffering to use
do not fucking add this, disarm mouth slapping SUCKED

**Comment by Tharcoonvagh** [comment]
Also the pr title is misleading

**Comment by Couls** [comment]
fair

**Comment by Wallemations** [comment]
I'm not a fan of this, simply due to the use of intents for emotes. As it's coded, it doesn't really add anything to combat whatsoever, so it shouldn't be tied to using disarm intent, as it will just make some people accidentally slap others when they want to shove them.

**Comment by anconfuzedrock** [comment]
do not add this it was terrible you'll have to trust me on the authority of literally having designed this code

**Comment by Fikou** [comment]
@Couls when will u close meme pr

**Comment by Couls** [comment]
the upvotes speak for themselves

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 1]
```diff
@@ -242,9 +242,21 @@
  * or another carbon.
 */
 /mob/living/carbon/proc/disarm(mob/living/carbon/target)
+	var/aimming_for_mouth  = zone_selected == BODY_ZONE_PRECISE_MOUTH
+	if(aimming_for_mouth)
```
Comment (bobbahbrown) [inline code/modules/mob/living/carbon/carbon_defense.dm:246]:
```suggestion
	if(zone_selected == BODY_ZONE_PRECISE_MOUTH)
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 2]
```diff
@@ -242,9 +242,21 @@
...
+		var/target_on_help_and_unarmed = target.a_intent == INTENT_HELP && !target.get_active_held_item()
+		var/target_restrained = target.restrained()
+		if(target_on_help_and_unarmed || target_restrained)
```
Comment (bobbahbrown) [inline code/modules/mob/living/carbon/carbon_defense.dm:249]:
```suggestion
		if(target_on_help_and_unarmed || target.restrained())
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 3]
```diff
@@ -242,9 +242,21 @@
...
+			do_slap_animation(target)
+			playsound(target.loc, 'sound/weapons/slap.ogg', 50, 1, -1)
+			visible_message("<span class='danger'>[src] slaps [target] in the face!</span>",
+				"<span class='notice'>You slap [target] in the face! </span>",\
+			"You hear a slap.")
+			if(target.dna && target.dna.species)
```
Comment (bobbahbrown) [inline code/modules/mob/living/carbon/carbon_defense.dm:255]:
```suggestion
			if(target.dna?.species)
```

File: code/modules/mob/living/living_defense.dm [hunk 4]
```diff
@@ -405,6 +405,18 @@
 	..()
 	setMovetype(movement_type & ~FLOATING) // If we were without gravity, the bouncing animation got stopped, so we make sure we restart the bouncing after the next movement.
 
+/mob/living/proc/do_slap_animation(atom/A)
```
Comment (bobbahbrown) [inline code/modules/mob/living/living_defense.dm:408]: Please document this proc to meet our [code documentation guidelines](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md#documenting-a-proc) for contributors.

**Comment by wesoda25** [comment]
If anything headpats should be made to work like the current slap, I'm not trying to accidentally headpat someone when I'm trying to help them up

**Comment by Couls** [comment]
can't believe stale bot is gonna kill this one

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 5]
```diff
@@ -240,9 +240,19 @@
  * or another carbon.
 */
 /mob/living/carbon/proc/disarm(mob/living/carbon/target)
+	if(zone_selected == BODY_ZONE_PRECISE_MOUTH)
+		var/target_on_help_and_unarmed = target.a_intent == INTENT_HELP && !target.get_active_held_item()
+		if(target_on_help_and_unarmed || target.restrained())
+			do_slap_animation(target)
+			playsound(target.loc, 'sound/weapons/slap.ogg', 50, 1, -1)
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/carbon_defense.dm:247]:
```suggestion
			playsound(target.loc, 'sound/weapons/slap.ogg', 50, TRUE, -1)
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 6]
```diff
@@ -240,9 +240,19 @@
...
+			visible_message("<span class='danger'>[src] slaps [target] in the face!</span>",
+				"<span class='notice'>You slap [target] in the face! </span>",\
+			"You hear a slap.")
+			if(target.dna?.species)
+				target.dna.species.stop_wagging_tail(target)
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/carbon_defense.dm:252]:
```suggestion
			if(target.dna?.species?.stop_wagging_tail(target)
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 7]
```diff
@@ -240,9 +240,18 @@
...
 /mob/living/carbon/proc/disarm(mob/living/carbon/target)
+	if(zone_selected == BODY_ZONE_PRECISE_MOUTH)
+		var/target_on_help_and_unarmed = target.a_intent == INTENT_HELP && !target.get_active_held_item()
+		if(target_on_help_and_unarmed || target.restrained())
+			do_slap_animation(target)
+			playsound(target.loc, 'sound/weapons/slap.ogg', 50, TRUE, -1)
+			visible_message("<span class='danger'>[src] slaps [target] in the face!</span>",
+				"<span class='notice'>You slap [target] in the face! </span>",\
+			"You hear a slap.")
+			if(target.dna?.species?.stop_wagging_tail(target)
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/carbon_defense.dm:251]:
```suggestion
			target.dna?.species?.stop_wagging_tail(target)
```

**Pull request merged**